checkstyle-tester/diff.groovy

```diff
@@ -623,36 +609,35 @@ def getProjectsStatistic(diffDir) {
     return projectsStatistic
 }
 
-def runMavenExecution(srcDir, excludes, checkstyleConfig,
-                      checkstyleVersion, extraMvnRegressionOptions) {
-    println "Running 'mvn clean' on $srcDir ..."
-    def mvnClean = "mvn -e --no-transfer-progress --batch-mode clean"
-    executeCmd(mvnClean)
-    println "Running Checkstyle on $srcDir ... with excludes {$excludes}"
-    def mvnSite = "mvn -e --no-transfer-progress --batch-mode site " +
-        "-Dcheckstyle.config.location=$checkstyleConfig -Dcheckstyle.excludes=$excludes"
-    if (checkstyleVersion) {
-        mvnSite = mvnSite + " -Dcheckstyle.version=$checkstyleVersion"
+def runCliExecution(allJar, srcDir, excludes, checkstyleConfig, saveDir,
+                      extraRegressionOptions) {
+    def saveLoc = new File(saveDir)
+    if (!saveLoc.exists()) {
+        saveLoc.mkdirs()
+    }
+
+    println "Running Checkstyle CLI on $srcDir ... with excludes {$excludes}"
+    def cliCommand = "java -jar $allJar -c $checkstyleConfig $srcDir -f xml -o " +
+        getOsSpecificPath("$saveDir", "checkstyle-result.xml") + " -e .git -e .hg"
```

Comment:
One of the benefits of using the CLI is we now pull in all files and not just `.java`. Because of this we have to exclude `.git` and `.hg` from erroneously processing them. `TreeWalker` will still trim to work only on Java files.